plugins/search-react/src/components/SearchBar/SearchBar.tsx

```diff
@@ -125,8 +119,8 @@ export const SearchBarBase: ForwardRefExoticComponent<SearchBarBaseProps> =
         }
       }, [onChange, onClear]);
 
-      const placeholder =
-        defaultPlaceholder ??
+      const label =
+        defaultLabel ??
```

Comment:
Do you mean I should keep the default placeholder as before and just have an optional label?

Reply:
Yes, exactly! What do you think?

Reply:
The issue with the placeholder is that it goes away as soon as the user inputs anything.

Reply:
The placeholder is still accessible to the user if they want to use it, it might be a breaking change but I do believe this is how it should work, will add a disclaimer to the changeset.

Reply:
This is expected behavior for a placeholder, so I don't see this as an issue... My intention is not to necessarily require everyone to use a label but to offer this possibility optionally for those who explicitly want to have a label (they can modify the default behavior by passing the label property).

Reply:
Our designer was the one to tell us to convert it to a label because a placeholder disappears on input, I can add support for both, the question is what to use as default for each, the current default do you want it for the placeholder or for the label?

Reply:
I wouldn't add a default value for the label, as we already have an aria-label with the value 'Search', but if one passes the label property explicitly, we can omit the aria-label.

Reply:
I removed the `aria-label` since you said having a `label` automatically adds the `aria-label`.

Reply:
Also the `aria-label` was ignoring the `label` prop, at this point we can:

1. Revert back to having only placeholder & an `aria-label`.
2. Add support for both `label` & `placeholder` & set up a default for both, removing the need for `aria-label`.
3. Same as .2 but label is optional, thus `aria-label` needs a default value, which only changes when a `label` is provided.
4. Same as .3 but the optional one is the `placeholder`, thus there's always a label whether the user sets it or just the default value.

Reply:
It is not automatically added, I said that we are adding it manually here: 
https://github.com/backstage/backstage/blob/f216fb8559c2abac1f6ab2307b7718c2e5e556ec/plugins/search-react/src/components/SearchBar/SearchBar.tsx#L157
It might not have been clear before, but the option I see is:
Add label and placeholder support and set up a placeholder and aria-label default value. If a label is passed, we do not use `aria-label`.

Reply:
I can commit the suggestion if it is a good idea 🙂

Reply:
I'm using the default value `Search in ${configApi.getOptionalString('app.title') || 'Backstage'}` as placeholder, and `Search` as `aria-label` default value if no label given.

Having some typing issues with `label` (react node) & `aria-label` (string), going to check if string & if not I'll ignore the `label`.